codegen/projections/white_label/spec/interceptor_spec.rb

```diff
@@ -4,25 +4,89 @@
 
 module WhiteLabel
   describe Client do
-    let(:test_interceptor) do
+    let(:interceptor_class) do
       Class.new do
+        def initialize(name='NA')
+          @name = name
+        end
         def read_before_execution(context); end
-      end.new
+      end
     end
 
+    let(:interceptor) { interceptor_class.new }
+    let(:hook) { :read_before_execution }
+
     describe 'client configured interceptor' do
       it 'calls interceptor hook' do
         config = Config.new(
           stub_responses: true,
-          interceptors: Hearth::InterceptorList.new([test_interceptor])
+          interceptors: Hearth::InterceptorList.new([interceptor])
         )
         client = Client.new(config)
 
-        expect(test_interceptor).to receive(:read_before_execution)
+        expect(interceptor).to receive(hook)
           .with(instance_of(Hearth::InterceptorContext))
 
         client.kitchen_sink()
       end
     end
+
+    describe 'operation configured interceptor' do
+      it 'calls interceptor hook' do
+        config = Config.new(
+          stub_responses: true
+        )
+        client = Client.new(config)
+
+        expect(interceptor).to receive(hook)
+                                 .with(instance_of(Hearth::InterceptorContext))
+
+        client.kitchen_sink({}, interceptors: [interceptor])
+      end
+    end
+
+    describe 'interceptor precedence' do
+      # * Interceptors registered via Class level plugins
+      # * Interceptors registered via config plugins (eg config.plugins)
+      # * Interceptors registered via config (eg, config.interceptors).
+      # * Interceptors registered via operation-level plugins.
+      # * Interceptors registered via operation-level options (passed directly).
+
+      let(:config_plugin_interceptor) { interceptor_class.new('cfg_plg') }
+      let(:config_interceptor) { interceptor_class.new('cfg_int') }
+      let(:operation_plugin_interceptor) { interceptor_class.new('op_plg') }
+      let(:operation_interceptor) { interceptor_class.new('op') }
+
+      let(:config_plugin) do
+        proc { |cfg| cfg.interceptors << config_plugin_interceptor}
```

Comment:
Space before `}` on this and below - we should enable rubocop styling/auto correct on spec files.

Reply:
ran rubocop on all of these.  But we should long term find a way to run it on all of these integration specs, but not on generated gems....